### ISSUE TITLE
Having a (transitive) PackageReference and a ProjectReference to the same assembly leads to an error

### ISSUE THREAD

**Issue by felixhirt** [dotnet/project-system#4663]
I have the following situation (everything with the new .csproj project system):
"Application" has 
   ProjectReference to "LibA"
   ProjectReference to "LibB"
and "LibA" has
   PackageReference to "LibB"

This causes a build error in VisualStudio (2019, RC4) because all the PackageReferences are not found. A manual "dotnet restore" shows that "An item with the same key has already been added. Key: (LibB, X.X.X)"

The error only happens if both versions are the same. If i increment the version of LibB e.g. in the ProjectReference, it works fine (and uses the newer version)

Searching did not yield much for me, i only found [4421](https://github.com/dotnet/project-system/issues/4421) which seems to be similar, but doesnt have much follow up (and no build error, thats why i opened a new issue)

If need be, i can try to create a simple reproduction solution

**Comment by jmarolf** [comment]
@felixhirt 	why doesn't `LibA` have a project reference to `LibB`?  Which dll do you actually want your application to use at runtime? The one from the nuget package or the one from the project you built?

**Comment by felixhirt** [comment]
Good questions. The situation is as follows: LibB is a basic building block which is referenced normally as a nuget package. There is a project we are currently working on where we decided to add the repo which contains the LibB source as a git submodule and reference the project directly in the final application. However, the same application has another submodule with a different set of library projects which partly build on top of LibB (as a nuget, because they are in different git repos), thats where LibA comes from. LibA and the other projects in there are usually referenced as projectreferences in large application projects, because they still receive lots of changes.

I guess the aim was to use the version of LibB that came from the projectreference for the final application, no matter what version. We just removed it now to make it work, there were no changes done to it anyway. If we had made changes, we could have easily gotten it to work by increasing the version of LibB locally, untill we would eventually release it as new nuget under that new version.

Now that i thought about the issue a litte more, i guess i would just like a better error message. Right now, the error is really elusive (the application did actually build for a long time, i guess because people added the project reference only after a while. I cannot tell which version of the dll it actually used in the end though (which might be a problem in itself!!). It only broke when a dev tried to add an additional nuget package, at which point suddenly Visual Studio showed a warning sign on all nugets, did not resolve them, and the build failed with weird errors about certain types not being found. Only after removing bin and obj folders did we see that the error was "an item with the same key has been added" when project.assets.json was generated, and only by running dotnet restore in the console did we see which reference caused the problem). Maybe it would be possible to show a build error like "your project reference to "LibB" (X.X.X) has an equal or lower version than the resolved version of the nuget for "LibB" (X.X.X). If you want to use the project reference of "LibB", increase its version. Otherwise remove one of the references"
